File: Php/app/User/SaltGenerator.php
namespace app\User{

	class SaltGenerator {

		private $saltLength = 10;
		private $allowedCharacters = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789_-';

		public function createSalt(){
			$stringGenerator = new RandomStringGenerator($this->saltLength, $this->allowedCharacters);
			return $stringGenerator->getString();
		}

	}

}

File: Core/Services/SaltGenerator.cs
namespace Core.Services{

	public class SaltGenerator : ISaltGenerator
	{
	    private readonly IRandomStringGenerator _randomStringGenerator;
	    private const int SaltLength = 10;
	    private const string AllowedCharacters = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789_-";

	    public SaltGenerator(IRandomStringGenerator randomStringGenerator)
	    {
	        _randomStringGenerator = randomStringGenerator;
	    }

	    public string CreateSalt(){
            return _randomStringGenerator.GetString(SaltLength, AllowedCharacters);
		}

	}

}